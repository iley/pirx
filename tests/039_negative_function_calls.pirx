func add(a: int, b: int): int {
  return a + b;
}

func negate(x: int): int {
  return -x;
}

extern func main(): int {
  var result: int;
  result = add(-5, -3);
  printf("add(-5, -3) = %d\n", result);
  result = negate(-10);
  printf("negate(-10) = %d\n", result);
  result = add(negate(5), negate(3));
  printf("add(negate(5), negate(3)) = %d\n", result);
  printf("Direct call: %d\n", add(-1, -2));
  return 0;
}