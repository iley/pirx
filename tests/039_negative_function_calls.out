add(-5, -3) = -8
negate(-10) = 10
add(negate(5), negate(3)) = -8
Direct call: -3
